projects/rocprofiler-sdk/source/lib/rocprofiler-sdk/aql/aql_profile_v2.h

```diff
@@ -446,6 +459,179 @@ aqlprofile_att_codeobj_marker(hsa_ext_amd_aql_pm4_packet_t*        packet,
+    AQLPROFILE_SPM_PARAMETER_TYPE_SAMPLE_INTERVAL,
+    AQLPROFILE_SPM_PARAMETER_TYPE_TIMEOUT,
+    AQLPROFILE_SPM_PARAMETER_TYPE_SAMPLE_MODE,
+    AQLPROFILE_SPM_PARAMETER_TYPE_LAST,
+} aqlprofile_spm_parameter_type_t;
+
+typedef enum
+{
+    AQLPROFILE_SPM_PARAMETER_SAMPLE_MODE_SCLK = 0,
+    AQLPROFILE_SPM_PARAMETER_SAMPLE_MODE_REFCLK
+} aqlprofile_spm_parameter_interval_mode_t;
+
+typedef struct
+{
+    aqlprofile_spm_parameter_type_t type;
+    uint64_t                        value;
+} aqlprofile_spm_parameter_t;
+
+/**
+ * @brief AQLprofile struct containing information for SPM counter events
+ */
+typedef struct
+{
+    aqlprofile_agent_handle_t     aql_agent;
+    hsa_agent_t                   hsa_agent;
+    const aqlprofile_pmc_event_t* events;
+    size_t                        event_count;
+    aqlprofile_spm_parameter_t*   parameters;
+    size_t                        parameter_count;
+    size_t                        reserved;  // For future use
+
+    aqlprofile_memory_alloc_callback_t
+        alloc_cb;  // Memory allocation, usually a wrapper for hsa_amd_memory_pool_allocate
+    aqlprofile_memory_dealloc_callback_t dealloc_cb;  // Frees memory allocated by alloc_cb
+    aqlprofile_memory_copy_t
+          memcpy_cb;  // Copy memory in and out of GPU memory allocated by alloc_cb
+    void* userdata;   // Passed back to user in the memory callbacks
+} aqlprofile_spm_profile_t;
+
+/**
+ * @brief Function to create control SPM packets
+ * @param[out] handle    To be passed to iterate_data()
+ * @param[out] desc      Used to decode SPM buffer contents
+ * @param[out] packets   Start/Stop AQL packets to be inserted in the queue
+ * @param[in] profile    Agent and events information
+ * @param[in] data_cb    Callback to retrieve SPM data when available
+ * @param[in] flags      Reserved. Must be zero.
+ * @param[in] userdata   Passed back to user
+ * @retval HSA_STATUS_SUCCESS on success
+ * @retval HSA_STATUS_ERROR   on generic error
+ * @retval HSA_STATUS_ERROR_OUT_OF_RESOURCES if memory allocation unsuccessful
+ * @retval HSA_STATUS_ERROR_INVALID_ARGUMENT for invalid parameter or event
+ * @retval HSA_STATUS_ERROR_INVALID_AGENT    for invalid agent handle
+ */
+hsa_status_t
+aqlprofile_spm_create_packets(aqlprofile_handle_t*          handle,
+                              aqlprofile_spm_buffer_desc_t* desc,
+                              aqlprofile_spm_aql_packets_t* packets,
+                              aqlprofile_spm_profile_t      profile,
+                              size_t                        flags);
+
+/**
+ * @brief Destroys resources allocated by aqlprofile_spm_create_packets()
+ * Implicitly calls aqlprofile_spm_stop. The descriptor pointer is invalid after this call.
+ * @param[in] handle Handle
+ */
+void
+aqlprofile_spm_delete_packets(aqlprofile_handle_t handle);
+
+typedef size_t aqlprofile_spm_buffer_handle_t;
+
+typedef enum
+{
+    AQLPROFILE_SPM_DATA_FLAGS_DATA_LOSS = 0,
```

Comment:
Are flags mutually exclusive values or we might end up with multiple flags set up at the same time?

Reply:
Wondering whether is more appropriate?
```suggestion
    AQLPROFILE_SPM_DATA_FLAGS_DATA_LOSS = 1 << 0,
```

Reply:
Rocprofiler flags: ROCPROFILER_SPM_RECORD_FLAG_DATA_LOST = 0 so that it is consistent with the flags in other modules